PROTO_tests/tests/pdarray_creation_test.py

```diff
@@ -0,0 +1,742 @@
+        assert np.arange(10, 0, -1).tolist() == ak.arange(10, 0, -1).to_list()
+        assert np.arange(-5, -10, -1).tolist() == ak.arange(-5, -10, -1).to_list()
+        assert np.arange(0, 10, 2).tolist() == ak.arange(0, 10, 2).to_list()
+
+    @pytest.mark.parametrize("dtype", ak.intTypes)
+    def test_arange_dtype(self, dtype):
+        # test dtype works with optional start/stride
+        stop = ak.arange(100, dtype=dtype)
+        assert np.arange(100, dtype=dtype).tolist() == stop.to_list()
+        assert dtype == stop.dtype
+
+        start_stop = ak.arange(100, 105, dtype=dtype)
+        assert np.arange(100, 105, dtype=dtype).tolist() == start_stop.to_list()
+        assert dtype == start_stop.dtype
+
+        start_stop_stride = ak.arange(100, 105, 2, dtype=dtype)
+        assert np.arange(100, 105, 2, dtype=dtype).tolist() == start_stop_stride.to_list()
+        assert dtype == start_stop_stride.dtype
+
+    def test_arange_misc(self):
+        # test uint64 handles negatives correctly
+        np_arange_uint = np.arange(2**64 - 5, 2**64 - 10, -1, dtype=np.uint64)
+        ak_arange_uint = ak.arange(-5, -10, -1, dtype=ak.uint64)
+        # np_arange_uint = array([18446744073709551611, 18446744073709551610, 18446744073709551609,
+        #        18446744073709551608, 18446744073709551607], dtype=uint64)
+        assert np_arange_uint.tolist() == ak_arange_uint.to_list()
+        assert ak.uint64 == ak_arange_uint.dtype
+
+        # test correct conversion to float64
+        np_arange_float = np.arange(-5, -10, -1, dtype=np.float64)
+        ak_arange_float = ak.arange(-5, -10, -1, dtype=ak.float64)
+        # array([-5., -6., -7., -8., -9.])
+        assert np_arange_float.tolist() == ak_arange_float.to_list()
+        assert ak.float64 == ak_arange_float.dtype
+
+        # test correct conversion to bool
+        expected_bool = [False, True, True, True, True]
+        ak_arange_bool = ak.arange(0, 10, 2, dtype=ak.bool)
+        assert expected_bool == ak_arange_bool.to_list()
+        assert ak.bool == ak_arange_bool.dtype
+
+        # test int_scalars covers uint8, uint16, uint32
+        uint_array = ak.arange(np.uint8(1), np.uint16(1000), np.uint32(1))
+        int_array = ak.arange(1, 1000, 1)
+        assert (uint_array == int_array).all()
+
+    @pytest.mark.parametrize("size", pytest.prob_size)
+    @pytest.mark.parametrize("array_type", [ak.int64, ak.float64, bool])
+    def test_randint_array_dtype(self, size, array_type):
+        test_array = ak.randint(0, size, size, array_type)
+        assert isinstance(test_array, ak.pdarray)
+        assert size == len(test_array)
+        assert array_type == test_array.dtype
+        assert [size] == test_array.shape
+        assert ((0 <= test_array) & (test_array <= size)).all()
+
+    @pytest.mark.parametrize("dtype", NUMERIC_SCALARS)
+    def test_randint_dtype(self, dtype):
+        for test_array in ak.randint(dtype(0), 100, 1000), ak.randint(0, dtype(100), 1000):
```

Comment:
we should add a comment here explaining why this is broken out from `test_randint_array_dtype` above. I know from offline convos that it's because some of the types (for example `np.uint8`) can overflow if `prob_size` gets large enough